nft/src/lib.rs

```diff
@@ -61,7 +61,7 @@ pub trait Config: frame_system::Config {
 	/// The token ID type
 	type TokenId: Parameter + Member + AtLeast32BitUnsigned + Default + Copy;
 	/// The class properties type
-	type ClassData: Parameter + Member;
+	type ClassData: Parameter + Member + Ord;
```

Comment:
I don't like this additional trait bound just so we can have it in genesis.

One way to avoid it is just use `Vec` instead of `BTreeMap`

Reply:
@xlc Hmm, it's probably not a good idea to restrict the type furthermore. However, I feel that the map is a really good fit in this situation (because it's a mapping of token classes to tokens themselves), is it possible to use another implementation that doesn't require `Ord`? Like this [one](https://substrate.dev/rustdocs/v2.0.0/hashbrown/struct.HashMap.html). On the other hand, introducing an external dependency doesn't sound like a good idea either. So something like this feel alright?

```
// each token will have an owner
pub type GenesisTokenData<T> = (T::AccountId, Vec<u8>, <T as Config>::TokenData);
// each token class will contain its tokens
pub type GenesisTokens<T> = (
	<T as frame_system::Config>::AccountId,
	Vec<u8>,
	<T as Config>::ClassData,
	Vec<GenesisTokenData<T>>
);
config(tokens): Vec<GenesisTokens<T>> {}
```

Reply:
Yep that should work.

Reply:
Fixed, I just wonder whether should I add any documentation for the types (in the form comments). Or is self-explanatory enough?

Reply:
some simple comment will be helpful, especially for `Vec<u8>` part

Reply:
Done